File: app/src/main/java/com/shoegazerwithak/lesswrongeveryday/ui/FragmentError.java
package com.shoegazerwithak.lesswrongeveryday.ui;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.shoegazerwithak.lesswrongeveryday.R;

public class FragmentError extends Fragment implements View.OnClickListener {
    private OnRetryListener mListener;

    public FragmentError() {
    }

    public static FragmentError newInstance() {
        return new FragmentError();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.error_screen_artists_list, container, false);
        Button buttonRetry = (Button) rootView.findViewById(R.id.button_retry);
        buttonRetry.setOnClickListener(this);
        return rootView;
    }

    public void setOnRetryListener(OnRetryListener listener) {
        mListener = listener;
    }

    @Override
    public void onClick(View v) {
        mListener.onRetry();
    }

    public interface OnRetryListener {
        void onRetry();
    }
}

File: app/src/main/java/com/shoegazerwithak/lesswrongeveryday/ui/FragmentError.kt
package com.shoegazerwithak.lesswrongeveryday.ui

import android.os.Bundle
import android.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button

import com.shoegazerwithak.lesswrongeveryday.R

class FragmentError : Fragment(), View.OnClickListener {
    private var mListener: OnRetryListener? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle): View? {
        val rootView = inflater.inflate(R.layout.error_screen_artists_list, container, false)
        val buttonRetry = rootView.findViewById(R.id.button_retry) as Button
        buttonRetry.setOnClickListener(this)
        return rootView
    }

    fun setOnRetryListener(listener: () -> Unit) {
        mListener = listener as OnRetryListener
    }

    override fun onClick(v: View) = mListener!!.onRetry()

    interface OnRetryListener {
        fun onRetry()
    }

    companion object {
        fun newInstance(): FragmentError = FragmentError()
    }
}